(jira-cli) init at 2.2

nyarly opened the pull request.

FRidh commented:
What is the reason for not using the packages in `python-packages.nix`?

nyarly commented:
There's no *reason* per se. I just used `pypi2nix` to build this - it's my first python nix expression.

Can you advise on how to use the python packages instead? My impression is of a lot of effort - is there gain to be had?

nyarly commented:
Ping on this PR?

FRidh commented:
`pyp2nix` can be used in Nixpkgs as a last resort. Otherwise the common packages should be used.

matthewbauer reviewed (dismissed):
See @fridh's comments.

nyarly commented:
@matthewbauer @FRidh I'll take a crack at the conversion.

nyarly commented:
@matthewbauer @FRidh I spent some time with this but am now stymied by `Could not find a version that satisfies the requirement argparse; python_version < "3.2" (from jira==1.0.10) (from versions: )`.

I see that argparse is nulled out intentionally in the pythonPackages expression, so I'm not sure how to proceed.

FRidh commented:
@nyarly argparse is part of the standard library of Python since 2.7. There is no reason for packages to any longer depend on it. The solution here is to patch the (typically) `setup.py` file to remove that dependency.

nyarly commented:
Okay, this is working with the distro python packages now.

FRidh commented on pkgs/top-level/python-packages.nix:
```diff
@@ -5097,6 +5097,26 @@ in {
     };
   };
 
+  keyrings = {
```
We are moving Python expressions out of `pkgs/top-level/python-packages.nix` into `pkgs/development/python-modules/<module>/default.nix`. 

Please move the expression there, and call it from `pkgs/top-level/python-packages.nix` using `callPackage ../development/python-modules/<package> { };`.

FRidh commented on pkgs/top-level/python-packages.nix:
```diff
@@ -5097,6 +5097,26 @@ in {
   };
 
+  keyrings = {
+    alt = buildPythonPackage rec {
```
This doesn't look good. It's supposed to be `keyrings-alt`. Don't nest it just so you get a dot.

FRidh commented on pkgs/top-level/python-packages.nix:
```diff
@@ -5097,6 +5097,26 @@ in {
+    alt = buildPythonPackage rec {
+      name = "keyrings.alt-${version}";
+      version = "2.3";
+      src = pkgs.fetchurl {
```
fetchPypi

FRidh commented on pkgs/top-level/python-packages.nix:
```diff
@@ -5097,6 +5097,26 @@ in {
+      propagatedBuildInputs = [ self.six ];
+
+      #
+      doCheck = false;
```
Why are the tests disabled? Please include a comment.

FRidh commented on pkgs/top-level/python-packages.nix:
```diff
@@ -5097,6 +5097,26 @@ in {
+      #
+      doCheck = false;
+
+      meta = with pkgs.stdenv.lib; {
```
maintainer?

FRidh replied:
homepage?

FRidh commented on pkgs/development/tools/jira_cli/default.nix:
```diff
@@ -0,0 +1,31 @@
+{ pkgs, stdenv, pythonPackages }:
```
Don't pass in `pkgs`, only the exact packages you need.

FRidh commented on pkgs/development/tools/jira_cli/default.nix:
```diff
@@ -0,0 +1,31 @@
+  inherit (pkgs) libffi openssl;
+  inherit (pythonPackages) fetchPypi buildPythonPackage;
+in
+  buildPythonPackage rec {
```
This is an application, use `buildPythonApplication`.

FRidh commented on pkgs/development/tools/jira_cli/default.nix:
```diff
@@ -0,0 +1,31 @@
+      substituteInPlace requirements/main.txt --replace "argparse" ""
+    '';
+
+    doCheck = false;
```
Why are the tests disabled? Please include a comment.

FRidh reviewed (changes requested):
The comments apply also to the other expressions.

And please read the contribution guidelines regarding Python packages.

FRidh commented on pkgs/top-level/python-packages.nix:
```diff
@@ -2993,6 +2993,23 @@ in {
 
   };
 
+  hiro = buildPythonPackage rec {
```
We are moving Python expressions out of `pkgs/top-level/python-packages.nix` into `pkgs/development/python-modules/<module>/default.nix`. 

Please move the expression there, and call it from `pkgs/top-level/python-packages.nix` using `callPackage ../development/python-modules/<package> { };`.

nyarly replied:
Aha - I'd misunderstood as applying to the keyrings set until you'd realized that alt was only a child to get the `.`. I'll take this on presently.

FRidh commented on pkgs/top-level/python-packages.nix:
```diff
@@ -5097,6 +5114,29 @@ in {
     };
   };
 
+  keyrings-alt = buildPythonPackage rec {
```
We are moving Python expressions out of `pkgs/top-level/python-packages.nix` into `pkgs/development/python-modules/<module>/default.nix`. 

Please move the expression there, and call it from `pkgs/top-level/python-packages.nix` using `callPackage ../development/python-modules/<package> { };`.

FRidh commented on pkgs/top-level/python-packages.nix:
```diff
@@ -5097,6 +5114,29 @@ in {
   };
 
+  keyrings-alt = buildPythonPackage rec {
+    name = "${pname}-${version}";
```
The `name` attribute is added by `buildPython*` and should therefore be removed.

FRidh commented on pkgs/top-level/python-packages.nix:
```diff
@@ -5097,6 +5114,29 @@ in {
+
+    # Fails with "ImportError: cannot import name mock"
+    doCheck = false;
+    #checkInputs = with self; [ pytest unittest2 mock keyring ];
```
This can be uncommented; they will be ignored when `doCheck = false;`

FRidh commented on pkgs/top-level/python-packages.nix:
```diff
@@ -6880,6 +6920,34 @@ in {
 
   jinja2_pluralize = callPackage ../development/python-modules/jinja2_pluralize { };
 
+  jira = buildPythonPackage rec {
```
We are moving Python expressions out of `pkgs/top-level/python-packages.nix` into `pkgs/development/python-modules/<module>/default.nix`. 

Please move the expression there, and call it from `pkgs/top-level/python-packages.nix` using `callPackage ../development/python-modules/<package> { };`.

FRidh commented on pkgs/top-level/python-packages.nix:
```diff
@@ -6880,6 +6920,34 @@ in {
   jinja2_pluralize = callPackage ../development/python-modules/jinja2_pluralize { };
 
+  jira = buildPythonPackage rec {
+    name = "${pname}-${version}";
```
The `name` attribute is added by `buildPython*` and should therefore be removed.

FRidh commented on pkgs/top-level/python-packages.nix:
```diff
@@ -7742,6 +7810,25 @@ in {
     };
   };
 
+  requests-oauthlib = buildPythonPackage rec {
```
We are moving Python expressions out of `pkgs/top-level/python-packages.nix` into `pkgs/development/python-modules/<module>/default.nix`. 

Please move the expression there, and call it from `pkgs/top-level/python-packages.nix` using `callPackage ../development/python-modules/<package> { };`.

FRidh commented on pkgs/top-level/python-packages.nix:
```diff
@@ -7742,6 +7810,25 @@ in {
   };
 
+  requests-oauthlib = buildPythonPackage rec {
+    name = "${pname}-${version}";
```
The `name` attribute is added by `buildPython*` and should therefore be removed.

FRidh commented on pkgs/development/tools/jira_cli/default.nix:
```diff
@@ -0,0 +1,32 @@
+
+    # Tests rely on VCR cassettes being written during tests. R/O nix store prevents this.
+    doCheck = false;
+    #checkInputs = with pythonPackages; [ vcrpy mock hiro ];
```
uncomment

nyarly commented:
@FRidh is there existing documentation about the move into `python-modules`? I think it probably makes sense, but AFAICT the header of `top-level/python-packages.nix` and the Python section of the manual don't refer to it (yet.) If not, is it okay if I issue a separate PR to that effect?

nyarly commented:
@FRidh How about now?

nyarly commented:
@FRidh Merged master (with alternative version of the jira package) into this branch. How's it look now?

FRidh commented:
> How about now?

Oh, I missed that.

Please don't do any merges, but instead rebase.

FRidh commented:
See also the contributing guidelines; one commit per logical change, which in that case would correspond to a commit per package.

nyarly commented:
@FRidh I can't find contribution guidelines about git usage within nixpkgs. I have to admit to some frustration about the process around adding this expression. I'm willing to wrap up with a rebase, but I'd like to know that I'm *done* before I do that.

nyarly commented:
@FRidh I appreciate your empathy. And I also understand that the number of PRs and issues outstrips the amount of time and attention available to review them. Thanks for all you're doing, and specifically for giving feedback to bring this home.

I've rebased and squashed this onto master. I hope it tells a clear narrative.

FRidh commented:
Still needed is a commit per package. You can squash your commits, do a soft reset, and then create a commit for each.

nyarly committed:
hiro: init at 0.1.4

nyarly committed:
keyrings.alt: init at 2.3

nyarly committed:
jira_cli: init at 2.2

nyarly commented:
@FRidh This is what you're looking for?

nyarly commented:
@FRidh Friendly reminder of this PR.

FRidh commented:
Yep, thanks. In the future please also prepend the package set the packages are defined in.
